### ISSUE TITLE
Issues with PowerShell v7.3.1 - Fixes Supplied

### ISSUE THREAD

**Issue by TwiggyDev** [ecitsolutions/Autotask#119]
Have run into some issues post-update to PowerShell v7.3.1; the AutoTask module was incurring a series of compiler errors.

2x Fixes are required:

**Reference.cs file (\AutoTask\2.0.1\Private\Reference.cs)**
For methods **OpenAsync()** and **CloseAsync()**, add the _new_ keyword in combination with the others. Example:
`public virtual new System.Threading.Tasks.Task OpenAsync()`
`public virtual new System.Threading.Tasks.Task CloseAsync()`

**Autotask.psm1 file (\AutoTask\2.0.1\Autotask.psm1)**
Change this section:
```
if ($PSVersionTable.PSVersion.Major -ge 6) {
    $assemblies += @(
        'netstandard'
        'System.Xml.XmlSerializer'
        'System.Runtime.Serialization.Xml'
        'System.ServiceModel.Primitives'
        'System.Private.ServiceModel'
        'System.Diagnostics.Tools'
    )
}
```

To:

```
if ($PSVersionTable.PSVersion.Major -ge 6) {
    $assemblies += @(
        'netstandard'
        'System.Xml.XmlSerializer'
        'System.Runtime.Serialization.Xml'
        'System.ServiceModel.Primitives'
        'System.Private.ServiceModel'
        'System.Diagnostics.Tools'
        'Microsoft.Bcl.AsyncInterfaces'
    )
}
```

Welcome :)

**Comment by AceOfNitwits** [comment]
Thank you!

**Comment by tmidthjell** [comment]
Thanks @TwiggyDev 
It would be great if this fix can be comitted and released as a minor patch 😊

**Comment by klemmestad** [comment]
Thank you, @TwiggyDev! I will include this into an update this week. I am trying not to change anything else, but there seem to have been thrown in a couple of minor changes that are throwing a couple of exceptions in my automatic rebuild system, but I expect to clear that out before the end of the week.

**Comment by klemmestad** [comment]
Module have been updated with a fix for this and updated to API version 1.6.11.
